predicators/refinement_estimators/base_refinement_estimator.py

```diff
@@ -16,8 +17,32 @@ def get_name(cls) -> str:
         use as the argument to `--refinement_estimator`."""
         raise NotImplementedError("Override me!")
 
+    @property
     @abc.abstractmethod
-    def get_cost(self, skeleton: List[_GroundNSRT],
+    def is_learning_based(self) -> bool:
+        """Does the estimator learn from training tasks?"""
+        raise NotImplementedError("Override me!")
+
+    @abc.abstractmethod
+    def get_cost(self, initial_state: State, skeleton: List[_GroundNSRT],
                  atoms_sequence: List[Set[GroundAtom]]) -> float:
         """Return an estimated cost for a proposed high-level skeleton."""
         raise NotImplementedError("Override me!")
+
+    def train(self, data: List) -> None:  # pragma: no cover
```

Comment:
we should be able to remove this and the below pragma-no-covers after we add some tests for the new learning-based class